pkgs/applications/networking/charles/default.nix

```diff
@@ -1,48 +1,79 @@
-{ stdenv, fetchurl, makeDesktopItem, jre, makeWrapper }:
-
 let
-  desktopItem = makeDesktopItem {
-    categories = "Network;Development;WebDevelopment;Java;";
-    desktopName = "Charles";
-    exec = "charles %F";
-    genericName  = "Web Debugging Proxy";
-    icon = "charles-proxy";
-    mimeType = "application/x-charles-savedsession;application/x-charles-savedsession+xml;application/x-charles-savedsession+json;application/har+json;application/vnd.tcpdump.pcap;application/x-charles-trace";
-    name = "Charles";
-    startupNotify = "true";
-  };
 
-in stdenv.mkDerivation rec {
-  name = "charles-${version}";
-  version = "4.2.7";
+  generic =
+      # dependencies
+      { stdenv, fetchurl, makeWrapper
+      , makeDesktopItem, jre
 
-  src = fetchurl {
-    url = "https://www.charlesproxy.com/assets/release/${version}/charles-proxy-${version}.tar.gz";
-    sha256 = "1nycw3wpbfwj4ijjaq5k0f4xipj8j605fs0yjzgl66gmv7r583rd";
-  };
+      # for overwriting
+      , this, self
```

Comment:
this and self are passed in but are not being used within the generic function. I think you're missing the [passthru block](https://github.com/NixOS/nixpkgs/blob/309c6d315bf6b5376ddc2f9d57a0c8b2ab1b4825/pkgs/servers/sql/postgresql/default.nix#L108-L122)

Reply:
Rewritten according to terraform :)